contracts/pair_stable/src/math.rs

```diff
@@ -26,12 +26,39 @@ pub fn calc_amount(balance_in: u128, balance_out: u128, amount_in: u128, amp: u6
 
     let d = compute_d(leverage, balance_in, balance_out).unwrap();
 
-    let new_balance_out = compute_new_balance_out(leverage, new_balance_in, d)?;
+    let new_balance_out = compute_new_balance(leverage, new_balance_in, d)?;
 
     let amount_swapped = balance_out - new_balance_out;
     Some(amount_swapped)
 }
 
+/// ## Description
+/// Calculates swapped amount.
+/// ## Params
+/// * **balance_in** is the object of type [`u128`].
+///
+/// * **balance_out** is the object of type [`u128`].
+///
+/// * **amount_out** is the object of type [`u128`].
+///
+/// * **amp** is the object of type [`u64`].
+pub fn calc_reverse_amount(
+    balance_in: u128,
```

Comment:
maybe we need to rename the parameters into `offer_pool` and `ask_pool`, `ask_amount` to represent the actual values that needs to be passed

Reply:
and maybe rename the function to `calc_offer_amount`, and do similar for `calc_amount` and rename it to `calc_ask_amount`

Reply:
Ok